swarm/storage/dpa.go

```diff
@@ -69,15 +69,15 @@ type DPA struct {
 // for testing locally
 func NewLocalDPA(datadir string, basekey []byte) (*DPA, error) {
 
-	hash := MakeHashFunc("SHA3")
+	params := NewLDBStoreParams(datadir, 0, nil, nil)
 
-	dbStore, err := NewLDBStore(datadir, hash, singletonSwarmDbCapacity, func(k Key) (ret uint8) { return uint8(Proximity(basekey[:], k[:])) })
+	dbStore, err := NewLDBStore(params)
```

Comment:
I see the need to reorganize various stores that are used currently, and there were discussions about that, but I am not sure that having constructors for store parameters is a bit strange. This leads to use something like this two lines. Where arguments to the NewLDBStore are wrapped by NewLDBStoreParems. I do not object it, as the complexity and coupling of stores is high, but in the future implementations, we could make it more idiomatic. I would not like to use this type of initialization, but given that the storage package types are coupled as they are this change is ok.